state_transfer/transfer/rsync/client.go

```diff
@@ -106,3 +136,30 @@ func createRsyncClient(c client.Client, r *RsyncTransfer, ns string) error {
 
 	return errorsutil.NewAggregate(errs)
 }
+
+// customizeTransportClientContainers customizes transport's client containers for specific rsync communication
+func customizeTransportClientContainers(t transport.Transport) {
```

Comment:
feels like we are not doing this right. One of the important functions of transport is to be able to set up containers for the transfer, this function virtually overwrites all the work done there.

Does anyone else feel weird about the division of responsibility here? @shawn-hurley @pranavgaikwad

Reply:
@alaypatel07 yeah, I felt that too. But I did this for the demo. We can talk about this in more detail.

Reply:
sure, let's chat, would be good to bounce ideas on enhancements here @pranavgaikwad